### PR TITLE
Add a temp project with Metric API (which will come eventually from runtime)

### PR DESCRIPTION
**Pull request by cijothomas**
Added a temp project to host metric API which is expected to come from .NET runtime.
Starting this to get started with a Otel Metric SDK implementation.

### CODE REVIEW

File: src/System.Diagnostics.Metrics.Temp/System.Diagnostics.Metrics.Temp.csproj [hunk 1]
```diff
@@ -0,0 +1,14 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <TargetFrameworks>netstandard2.0;net452</TargetFrameworks>
```
Comment (bruno-garcia) [inline src/System.Diagnostics.Metrics.Temp/System.Diagnostics.Metrics.Temp.csproj:3]: Is this planned to get published to NuGet?
Assuming it's not, might want to throw:

```
<PropertyGroup>
  <IsPackable>false</IsPackable>
</PropertyGroup>
```

Assuming `dotnet pack` will run on the whole project and pushed somewhat automatically to nuget.org

File: src/System.Diagnostics.Metrics.Temp/CounterFunc.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+#nullable enable
+
+namespace System.Diagnostics.Metrics
+{
+    public class CounterFunc<T> : ObservableMeterInstrument<T> where T : unmanaged
```
Comment (reyang) [inline src/System.Diagnostics.Metrics.Temp/CounterFunc.cs:11]: I think this is called `ObservableCounter` (or do you intend to change it later)?

Comment (cijothomas) [reply src/System.Diagnostics.Metrics.Temp/CounterFunc.cs:11]: yes will be change to match https://github.com/dotnet/designs/pull/211/files. This PR is to get started with a temp proj file.

File: src/System.Diagnostics.Metrics.Temp/System.Diagnostics.Metrics.Temp.csproj (hunk 1 above)
Comment (cijothomas) [reply src/System.Diagnostics.Metrics.Temp/System.Diagnostics.Metrics.Temp.csproj:3]: Will do! (we publish to myget automatically, and dont want to publish this)

File: src/System.Diagnostics.Metrics.Temp/Distribution.cs [hunk 3]
```diff
@@ -0,0 +1,29 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace System.Diagnostics.Metrics
+{
+    public class Distribution<T> : MeterInstrument<T> where T : unmanaged
```
Comment (reyang) [inline src/System.Diagnostics.Metrics.Temp/Distribution.cs:9]: There are many places not updated to the latest API spec:
1. Instrument names (Distribution -> Histogram, Gauge -> ObservableGauge).
2. Label - Attribute (or Tag?)

**Review by reyang — APPROVED** [review]
There are some out-dated stuff, but I think all of them can be addressed by follow up PRs.

**Pull request merged**